flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -1046,6 +1047,8 @@ void failGlobalIfExecutionIsStillRunning(Throwable cause, ExecutionAttemptID fai
 	 * @param t The exception that caused the failure.
 	 */
 	public void failGlobal(Throwable t) {
+		ClusterEntryPointExceptionUtils.tryEnrichClusterEntryPointError(t);
```

Comment:
I am not sure whether we can enrich `t` here because `t` can also come from the `TaskExecutor` if a `Task` has failed. In this case, the ClusterEntryPoint specific messages don't make sense.

Reply:
The OOM handling was moved into `ExecutionGraph.vertexFinished()`